docs/advanced-usage/using-proxies.mdx

```diff
@@ -102,6 +102,43 @@ When using a proxy, all requests to the Frontend API will be made through your d
         ```
       </CodeBlockTabs>
     </Tab>
+
+    <Tab>
+      ```ts {{ prettier: false, filename: 'middleware.ts' }}
+        import { NextResponse } from "next/server";
+        import { clerkMiddleware } from "@clerk/nextjs/server";
+
+        export default clerkMiddleware((auth, request) => {
+          if (request.url.match("__clerk")) {
+            const proxyHeaders = new Headers(request.headers);
+            proxyHeaders.set("Clerk-Proxy-Url", process.env.NEXT_PUBLIC_CLERK_PROXY_URL || "");
+            proxyHeaders.set("Clerk-Secret-Key", process.env.CLERK_SECRET_KEY || "");
+            proxyHeaders.set("X-Forwarded-For", request.ip || "");
```

Comment:
might also need to support cases where nextjs is self-hosted

https://nextjs.org/docs/app/api-reference/functions/next-request#ip

Reply:
Good call!